runtime/runtime-params-estimator/src/action_costs.rs

```diff
@@ -0,0 +1,626 @@
+        .apply_cost(&mut ctx.testbed())
+        / ActionSize::Max.deploy_contract()
+}
+
+pub(crate) fn function_call_base_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Min))
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn function_call_base_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Min))
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn function_call_base_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Min))
+        .apply_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn function_call_byte_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Max))
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+        / ActionSize::Max.function_call_payload()
+}
+
+pub(crate) fn function_call_byte_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Max))
+        .verify_cost(&mut ctx.testbed())
+        / ActionSize::Max.function_call_payload()
+}
+
+pub(crate) fn function_call_byte_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(function_call_action(ActionSize::Max))
+        .apply_cost(&mut ctx.testbed())
+        / ActionSize::Max.function_call_payload()
+}
+
+pub(crate) fn transfer_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(transfer_action())
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn transfer_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx).add_action(transfer_action()).verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn transfer_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx).add_action(transfer_action()).apply_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn stake_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(stake_action())
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SameAsSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn stake_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(stake_action())
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::RandomUnused)
```

Comment:
Yes, superfluous, I removed it.
For the second question, I now added a comment:
```
/// This is not a useful action, as staking only works with sender = receiver.
/// But since this fails only in the exec step, we must still charge a fitting
/// amount of gas in the send step.
```